### ISSUE TITLE
Template file `No bake template found for "fixture_factory" skipping file generation.` could not be found

### ISSUE THREAD

**Issue by LordSimal** [pakacuda/cakephp-fixture-factories#2]
## What is the error?
```
bin/cake bake fixture_factory -a -f
Exception: Template file `No bake template found for "fixture_factory" skipping file generation.` could not be found.
In [No bake template found for "fixture_factory" skipping file generation., line 104]

2020-05-01 11:19:41 Error: [Cake\View\Exception\MissingTemplateException] Template file `No bake template found for "fixture_factory" skipping file generation.` could not be found. in No bake template found for "fixture_factory" skipping file generation. on line 104
Exception Attributes: array (
  'file' => 'No bake template found for "fixture_factory" skipping file generation.',
  'paths' => 
  array (
  ),
)
Stack Trace:
- <my-path>/vendor/pakacuda/cakephp-fixture-factories/src/Command/FixtureFactoryCommand.php:198
- <my-path>/vendor/pakacuda/cakephp-fixture-factories/src/Command/FixtureFactoryCommand.php:139
- <my-path>/vendor/pakacuda/cakephp-fixture-factories/src/Command/FixtureFactoryCommand.php:168
- <my-path>/vendor/cakephp/cakephp/src/Console/BaseCommand.php:175
- <my-path>/vendor/cakephp/cakephp/src/Console/CommandRunner.php:336
- <my-path>/vendor/cakephp/cakephp/src/Console/CommandRunner.php:171
- <my-path>/bin/cake.php:12
```

## What did I do?
```
composer require --dev pakacuda/cakephp-fixture-factories  
```

This successfully added the plugin to the vendor
```
composer require --dev pakacuda/cakephp-fixture-factories    
Using version ^0.2.4 for pakacuda/cakephp-fixture-factories
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 2 installs, 0 updates, 0 removals
  - Installing fzaninotto/faker (v1.9.1): Downloading (100%)         
  - Installing pakacuda/cakephp-fixture-factories (v0.2.4): Downloading (100%)         
Writing lock file
Generating autoload files
> Cake\Composer\Installer\PluginInstaller::postAutoloadDump
17 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
```

Then i manually added the Plugin into the **bootstrapCli()** function in the **src/Application.php**
```
$this->addPlugin('CakephpFixtureFactories');
```

After that I executed
```
bin/cake bake fixture_factory -a -f   
```

For me it seems like it can't find the **vendor/pakacuda/cakephp-fixture-factories/templates/bake/fixture_factory.twig** even though it is present.

<details>
<summary>My composer.json</summary>

```
{
  "name": "cakephp/app",
  "description": "CakePHP skeleton app",
  "homepage": "https://cakephp.org",
  "type": "project",
  "license": "MIT",
  "require": {
    "php": ">=7.2",
    "ext-curl": "*",
    "ext-ftp": "*",
    "ext-json": "*",
    "cakephp/cakephp": "^4.0",
    "cakephp/migrations": "3.0.0-beta2",
    "cakephp/plugin-installer": "^1.0",
    "gamez/mite": "^1.0",
    "guzzlehttp/guzzle": "^6.5",
    "jeroendesloovere/vcard": "^1.7",
    "mobiledetect/mobiledetectlib": "2.*",
    "phan/phan": "^2.4",
    "phpoffice/phpspreadsheet": "^1.9"
  },
  "require-dev": {
    "cakephp/bake": "^2.0",
    "cakephp/cakephp-codesniffer": "dev-next",
    "cakephp/debug_kit": "^4.0",
    "dereuromark/cakephp-ide-helper": "^1.5",
    "easybill/php-sdk": "^2.0",
    "josegonzalez/dotenv": "3.*",
    "pakacuda/cakephp-fixture-factories": "^0.2.4",
    "phpseclib/phpseclib": "^2.0",
    "phpunit/phpunit": "^8.0",
    "psy/psysh": "@stable"
  },
  "suggest": {
    "markstory/asset_compress": "An asset compression plugin which provides file concatenation and a flexible filter system for preprocessing and minification.",
    "dereuromark/cakephp-ide-helper": "After baking your code, this keeps your annotations in sync with the code evolving from there on for maximum IDE and PHPStan compatibility."
  },
  "autoload": {
    "psr-4": {
      "App\\": "src/"
    }
  },
  "autoload-dev": {
    "psr-4": {
      "App\\Test\\": "tests/",
      "Cake\\Test\\": "vendor/cakephp/cakephp/tests/"
    }
  },
  "scripts": {
    "post-install-cmd": "App\\Console\\Installer::postInstall",
    "post-create-project-cmd": "App\\Console\\Installer::postInstall",
    "post-autoload-dump": "Cake\\Composer\\Installer\\PluginInstaller::postAutoloadDump",
    "check": [
      "@test",
      "@cs-check"
    ],
    "cs-check": "phpcs --colors -p --standard=vendor/cakephp/cakephp-codesniffer/CakePHP src/ tests/",
    "cs-fix": "phpcbf --colors --standard=vendor/cakephp/cakephp-codesniffer/CakePHP src/ tests/",
    "stan": "phpstan analyse src/",
    "stan-setup": "cp composer.json composer.backup && composer require --dev phpstan/phpstan-shim:^0.11 && mv composer.backup composer.json",
    "test": "phpunit --colors=always"
  },
  "prefer-stable": true,
  "config": {
    "sort-packages": true
  },
  "minimum-stability": "dev"
}
```
</details>

<details>
<summary>My phpunit.xml</summary>

```
<?xml version="1.0" encoding="UTF-8"?>
<phpunit
  colors="true"
  bootstrap="tests/bootstrap.php"
  backupGlobals="true"
>

  <testsuites>
    <testsuite name="cakephp">
      <directory>tests/TestCase/</directory>
      <!-- Excludes are required in order to let DatabaseSuite decorate the tests -->
      <exclude>tests/TestCase/Database/</exclude>
      <exclude>tests/TestCase/ORM/</exclude>
    </testsuite>
    <testsuite name="database">
      <file>tests/TestCase/DatabaseSuite.php</file>
    </testsuite>
  </testsuites>

  <listeners>
    <listener class="Cake\TestSuite\Fixture\FixtureInjector">
      <arguments>
        <object class="Cake\TestSuite\Fixture\FixtureManager"/>
      </arguments>
    </listener>
  </listeners>

  <!-- Setup a listener for fixtures -->
  <listeners>
    <listener class="CakephpFixtureFactories\TestSuite\FixtureInjector">
      <arguments>
        <object class="CakephpFixtureFactories\TestSuite\FixtureManager" />
      </arguments>
    </listener>
  </listeners>

  <!-- Prevent coverage reports from looking in tests, vendors, config folders -->
  <filter>
    <whitelist>
      <directory suffix=".php">src/</directory>
      <exclude>
        <!--
        This file contains a few functions that cannot be tested
        as they contain die; or breakpoint functionality.
        -->
        <file>src/basics.php</file>
        <!-- This file is deprecated and unused -->
        <file>src/Core/ClassLoader.php</file>
      </exclude>
    </whitelist>
  </filter>

  <php>
    <ini name="memory_limit" value="-1"/>
    <!-- E_ALL & ~E_USER_DEPRECATED (16383)-->
    <!-- E_ALL (32767) -->
    <ini name="error_reporting" value="32767"/>

    <!-- SQLite
    <env name="DB_DSN" value="sqlite:///:memory:"/>
    -->
    <!-- Postgres
    <env name="DB_DSN" value="postgres://localhost/cake_test?timezone=UTC"/>
    -->
    <!-- Mysql
    <env name="DB_DSN" value="mysql://localhost/cake_test?timezone=UTC"/>
    -->
    <!-- SQL Server
    <env name="DB_DSN" value="sqlserver://localhost/cake_test?timezone=UTC"/>
    -->
  </php>
</phpunit>
```
</details>

**Comment by pabloelcolombiano** [comment]
Hi,

thanks for giving it a try. Are you using CakePHP 3.x or 4.x?

For 3.x use "^0.1.0", 4.x use "^0.2.0"

I'll update the documentation.

Let me know if this fixed you issue!

**Comment by LordSimal** [comment]
As you can see in my composer.json in the Issue I use cakephp/cakephp:^4.0

 

To be specific I use 4.0.6

 

Von: webrider <notifications@github.com> 
Gesendet: Freitag, 1. Mai 2020 15:47
An: pakacuda/cakephp-fixture-factories <cakephp-fixture-factories@noreply.github.com>
Cc: Kevin Pfeifer <info@pfiff.me>; Author <author@noreply.github.com>
Betreff: Re: [pakacuda/cakephp-fixture-factories] Template file `No bake template found for "fixture_factory" skipping file generation.` could not be found (#2)

 

Hi,

thanks for giving it a try. Are you using CakePHP 3.x or 4.x?

For 3.x use "^0.1.0", 4.x use "^0.2.0"

I'll update the documentation.

Let me know if this fixed you issue!

**Comment by pabloelcolombiano** [comment]
@LordSimal indeed, I saw only after that you had given your full composer.

Well first thanks for trying out the vendor, and for your feedback. I habe applied a fix.
Would you mind trying a gain, reinstalling the vendor?

**Comment by LordSimal** [comment]
Since composer needs some time to update I just applied your change in the vendor/pakacuda/cakephp-fixture-factories/src/Command/FixtureFactoryCommand.php manually and now the command works and the factories got generated!

Thanks for that quick fix!

PS: I found your repository because you were in the "CakePHP Online Meetup 2020-04 with Mark Scherer" stream 😄 
And since I have a CakePHP project with 44 tables and basically no testing I was quite intrigued by your factory.
